### PR TITLE
Transitions for Great Win!

### PR DESCRIPTION
**Pull request by leostera**

### CODE REVIEW

**Comment by leostera** [comment]
Please don't merge just yet! Still need to:
- [ ] Clean up the hardcoded values and make them configurable as Slide props
- [x] Clean up the sample project to not install unveil using npm but actually rely on the sources already provided (great as a development presentation)
- [ ] Figure out how to define the animations without having to go `.slide .slide-content` – I think we can do better than this.

File: src/components/UnveilApp.js [hunk 1]
```diff
@@ -36,6 +36,13 @@ export default React.createClass({
     }).compact();
   },
 
+  autoKeySlides: function (children) {
+    if (!Array.isArray(children) || children.length === 0) return children;
+    return React.Children.map(children, function (c) {
+      return React.cloneElement(c, { key: c.props.name || '' }, this.autoKeySlides(c.props.children));
```
Comment (irisSchaffer) [inline src/components/UnveilApp.js]: what about slides without a name?

File: src/helpers/History.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import {createHashHistory, useQueries} from 'history';
+import {createHistory, useQueries} from 'history';
 
-let history = useQueries(createHashHistory)({
+let history = useQueries(createHistory)({
```
Comment (irisSchaffer) [inline src/helpers/History.js]: brill!

File: src/components/UnveilApp.js (hunk 1 above)
Comment (leostera) [inline src/components/UnveilApp.js]: They'd just key the `/0.{n}` added, which would still make them unique

**Pull request merged**